moveit_ros/hybrid_planning/local_planner/local_constraint_solver_plugins/src/forward_trajectory.cpp

```diff
@@ -108,7 +114,12 @@ ForwardTrajectory::solve(const robot_trajectory::RobotTrajectory& local_trajecto
     {
       planning_scene_monitor::LockedPlanningSceneRO locked_planning_scene(planning_scene_monitor_);
```

Comment:
we should probably update the current state before checking using `updateSceneWithCurrentState()` as well, that also takes care about updating transforms. After that we can directly call [PlanningScene::checkCollision()](https://moveit.picknik.ai/humble/api/html/classplanning__scene_1_1PlanningScene.html#a4ebcc755346e443319104e8b964f8faa) So that we don't have to copy the current state. It would be even better if we could extend the [distanceToCollision()](https://moveit.picknik.ai/humble/api/html/classplanning__scene_1_1PlanningScene.html#a838cbad712afd5bc66374f4bb46402fa) API to support groups and current state

Reply:
done